edge-util/src/Microsoft.Azure.Devices.Edge.Storage/MemorySpaceChecker.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage
+{
+    using System;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class MemorySpaceChecker : IStorageSpaceChecker
+    {
+        internal enum MemoryUsageStatus
+        {
+            Unknown = 0,
+            Available,
+            Critical,
+            Full
+        }
+
+        long maxSize;
+        bool isEnabled;
+        Func<long> getTotalMemoryUsage;
+        MemoryUsageStatus memoryUsageStatus;
+
+        public MemorySpaceChecker(Func<long> getTotalMemoryUsage)
+        {
+            this.getTotalMemoryUsage = Preconditions.CheckNotNull(getTotalMemoryUsage, nameof(getTotalMemoryUsage));
+            this.maxSize = long.MaxValue;
+            this.memoryUsageStatus = MemoryUsageStatus.Unknown;
+        }
+
+        public void SetMaxSizeBytes(Option<long> maxSizeBytes)
+        {
+            this.maxSize = maxSizeBytes.Match(
+                s =>
+                {
+                    this.isEnabled = true;
+                    Events.SetMaxMemorySpaceUsage(s);
```

Comment:
This gets set when there is a twin update with a config change, right? I think it should be thread safe

Reply:
I don't think this needs to be thread-safe given how ConfigUpdater takes locks before calling these methods which should suffice. I can add a comment on the method to indicate that the method isn't supposed to be thread-safe.